code/game/objects/items.dm

```diff
@@ -680,10 +724,11 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 /obj/item/proc/dropped(mob/user, silent = FALSE)
 	SHOULD_CALL_PARENT(TRUE)
-	for(var/X in actions)
-		var/datum/action/A = X
-		A.Remove(user)
-
+	// Remove any item actions we temporary gave out.
+	for(var/datum/action/action_item_has as anything in actions)
+		action_item_has.Remove(user)
+	if(item_flags & DROPDEL)
+		qdel(src)
```

Comment:
```suggestion
		action_item_has.Remove(user)
```

Reply:
This is a duplicate, we already handle DROPDEL at the bottom of the proc.

This is whats causing signal runtimes and your qdel loop

Reply:
Fuck